drivers/i2c/Kconfig

```diff
@@ -135,6 +135,13 @@ config I2C_MCUX_LPI2C
 	help
 	  Enable the mcux LPI2C driver.
 
+config I2C_MCUX_LPI2C_SPEED_HIGH
```

Comment:
I would rather see this implemented as a DT property, since support for high speed mode is a property of the hardware, not a software setting. This should also make it easier to add support for high speed mode on any future SOCs that support it.

Reply:
Actually, there is DT property `clock-frequency`  to seclect speed mode. We need to set `I2C_BITRATE_HIGH` for high speed mode. But suppose users don't know the exact supported speed mode, eg. mcux lpi2c support max speed is `I2C_SPEED_FAST_PLUS`, but that  they set high speed, there will be not checking mechanism for rasing error. so I added this kconfig to wrap support high speed mode that depends on SoCs (current only SoC K3, more add depend on for new SoCs).

Reply:
It would be better to return an error if the device doesnt support a mode rather than make Kconfigs like this. I suppose the other modes suffer from the same potential problem.

Reply:
>I suppose the other modes suffer from the same potential problem.
it seems you say right. This potential problem is not only for high speed mode. So I 
removed the kconfig.

Reply:
I'd like to clarify something here: does the LPI2C peripheral on the S32 support fast mode in controller mode, or only in target mode? The RM suggests support is only available in target mode, but this API is not used for that case

Reply:
I double checked, the high speed mode is only supported in target mode. And that's right this API is not used for target mode. So I removed my change.